### PR TITLE
added todo_app files

### PR DESCRIPTION
**Pull request by BhujelVictor**
# Pull Request

## Description
[Added ToDo app files]

## Checklist
- [ ] Created models ToDoList and ToDoItem and defined a one-to-many relationship for each list  item to be always associated with an item list.
- [ ] Created two views ListListView and ItemListView by subclassing django.views.generic.ListView for Read operation
- [ ] Created two templates index.html to show toto list and todo_list.html to show todo list items with associated date both inherits from base.html 

## Additional Notes
[Need to add remaining CRUD views(create, update and delete)  and other additional features\.]

demo_1
demo2

### CODE REVIEW

File: TODO/TODO/settings.py [hunk 1]
```diff
@@ -124,7 +125,11 @@
 # ]
 
 
-AUTHENTICATION_BACKENDS = ['authuser.backends.EmailBackend']
+AUTHENTICATION_BACKENDS = [
+    'authuser.backends.EmailBackend',
+    'django.contrib.auth.backends.ModelBackend'
```
Comment (yonjans) [inline TODO/TODO/settings.py:130]: Check how does it work with multiple authentication backends.

File: TODO/todo_app/models.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+# to add a default due time while creating a todo item
+def one_week_hence():
+    return timezone.now() + timezone.timedelta(days=7)
+
+#defining title of the list
+class ToDoList(models.Model):
+    title = models.CharField(max_length=100, unique=True)
+
+    def get_absolute_url(self):
```
Comment (yonjans) [inline TODO/todo_app/models.py:14]: What will this function do?

File: TODO/todo_app/models.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    title = models.CharField(max_length=100)
+    description = models.TextField(null=True, blank=True)
+    created_date = models.DateTimeField(auto_now_add=True)
+    due_date = models.DateTimeField(default=one_week_hence)
+    # Foreign key: each to-do item must belong to a to-do list
+    # on_delete: deletes the all the to-do items associated with a to-do list
+    todo_list = models.ForeignKey(ToDoList, on_delete=models.CASCADE)
+
+
+    def get_absolute_url(self):
```
Comment (yonjans) [inline TODO/todo_app/models.py:31]: Same
> What will this function do?

File: TODO/todo_app/templates/todo_app/index.html [hunk 4]
```diff
@@ -0,0 +1,24 @@
+{% extends "base.html" %}
+
+{% block title %}And then the list where bestowed upon us by thee {% endblock %}
+{% block content %}
+
+{% if object_list %}
```
Comment (yonjans) [inline TODO/todo_app/templates/todo_app/index.html:6]: Where will this `object_list` be populated?

File: TODO/todo_app/templates/todo_app/todo_list.html [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+<div>
+
+    <div>
+
+        <div>
+
+            <h3>Edit list:</h3>
+
+            <h5>{{ todo_list.title | upper | linebreaksbr }}</h5>
```
Comment (yonjans) [inline TODO/todo_app/templates/todo_app/todo_list.html:13]: Where will this `todo_list` be populated?

File: TODO/todo_app/templates/todo_app/todo_list.html [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+        </div>
+
+        <ul>
+
+            {% for todo in object_list %}
```
Comment (yonjans) [inline TODO/todo_app/templates/todo_app/todo_list.html:19]: Where will this `object_list` be populated?

**Review by yonjans — CHANGES_REQUESTED** [review]

File: TODO/TODO/settings.py [hunk 7]
```diff
@@ -38,6 +38,7 @@
     'django.contrib.messages',
     'django.contrib.staticfiles',
     'authuser',
+    'todo_app',
```
Comment (yonjans) [inline TODO/TODO/settings.py:41]: Also need a recording on how it works.

**Review by yonjans — CHANGES_REQUESTED** [review]

File: TODO/TODO/settings.py (hunk 1 above)
Comment (BhujelVictor) [reply TODO/TODO/settings.py:130]: **Backends authentication working principle:**
- mulltiple backends can be used in a tandem 
- is authenticated on the first backend check if found

File: TODO/todo_app/models.py (hunk 2 above)
Comment (BhujelVictor) [reply TODO/todo_app/models.py:14]: Class based view to create models in database:
-Creates a table named todolist with a field that stores characters named title and with the defined params as above

File: TODO/todo_app/models.py (hunk 3 above)
Comment (BhujelVictor) [reply TODO/todo_app/models.py:31]: get_absolute_url(): Tells the django how to calculate the canonical URL for an object, returns a string that can be used to refer to an object over the HTTP.
For example:

def get_absolute_url(self):
        return reverse("list", args=[self.id])


Here,
      this function returns the URL for the list item refering to the id of the object.

**Pull request merged**